### DISCUSSION TITLE
useMutation  : TS2769: No overload matches this call.   The last overload gave the following error.

### DISCUSSION THREAD

**Discussion by ghost** [TanStack/query#2350] [Q&A]
`TS2769: No overload matches this call.   The last overload gave the following error.     Argument of type '(a: any, b: any, c: any) => Promise<any>' is not assignable to parameter of type 'MutationKey'.       Type '(a: any, b: any, c: any) => Promise<any>' is missing the following properties from type 'readonly unknown[]': concat, join, slice, indexOf, and 17 more.`

![image](https://user-images.githubusercontent.com/6580897/120982126-f0029180-c7aa-11eb-8afc-0a03a5688ae5.png)
![image](https://user-images.githubusercontent.com/6580897/120982172-fc86ea00-c7aa-11eb-90b0-ed41622cc57b.png)

**Comment by TkDodo** [comment]
`useMutation` only takes _one_ argument for input variables. If you want more, pleas use an object:

```
type Variables = {
  current_password: string,
  new_password: string,
  re_new_password: string
}

function password_change(variables: Variables): Promise<any>

useMutation(password_change)

```

```
